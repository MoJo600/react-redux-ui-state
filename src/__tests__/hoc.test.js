import 'jsdom-global/register';
import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { uiState } from '../';

configure({ adapter: new Adapter() });

describe('uiState hoc', () => {
    function testComponent({ display }) {
        console.log('testComponent'); // eslint-disable-line
        return (
            <div>
                Display: {display.toString()}
            </div>
        );
    }

    testComponent.propTypes = {
        display: PropTypes.bool,
    };

    const uiStateConfig = {
        name: 'test-component',
        state: () => ({
            display: false,
        }),
    };

    it('should do render without exploding', () => {
        const component = (
            <Provider store={createStore(uiState)}>
                <wrapped-test-component />
            </Provider>
        );

        const wrappedTestComponent = uiState(uiStateConfig)(testComponent);
        const renderedComponent = mount(component);
        const tree = toJson(renderedComponent);

        expect(tree).toMatchSnapshot();
        expect(wrappedTestComponent).toMatchSnapshot();
    });
});
