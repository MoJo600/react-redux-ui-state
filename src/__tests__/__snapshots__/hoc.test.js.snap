// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiState hoc should do render without exploding 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <wrapped-test-component />
</Provider>
`;

exports[`uiState hoc should do render without exploding 2`] = `[Function]`;
